Userland/Libraries/LibCore/Group.cpp

```diff
@@ -60,6 +62,29 @@ ErrorOr<void> Group::add_group(Group& group)
 }
 #endif
 
+ErrorOr<Vector<Group>> Group::all()
+{
+    Vector<Group> groups;
+
+    ScopeGuard grent_guard([] { endgrent(); });
+    setgrent();
+    errno = 0;
+    for (auto const* gr = getgrent(); gr; gr = getgrent()) {
+        if (errno)
+            return Error::from_errno(errno);
```

Comment:
How would `errno` be non-zero here? If `getgrent()` returns null, we will exit the loop. Seems like this check should be after the loop rather than inside it :)